snakefile.py

```diff
@@ -193,11 +202,34 @@ def tool(name):
     shell: "{SAMTOOLS_EXEC} index {input} {output} >> {log} 2>&1"
 
 
-rule fastqc:
-    input: os.path.join(MAPPED_READS_DIR, '{sample}_aligned_sorted.bam')
-    output: os.path.join(FASTQC_DIR, '{sample}_aligned_sorted_fastqc.zip')
-    log: os.path.join(LOG_DIR, 'fastqc_{sample}.log')
-    shell: "{FASTQC_EXEC} -o {FASTQC_DIR} -f bam {input} >> {log} 2>&1"
+rule fastqc_raw:
+    input: os.path.join(READS_DIR, "{sample}_R{read_num}.fastq")
+    output: os.path.join(FASTQC_DIR, '{sample}', '{sample}_R{read_num}_fastqc.zip')
+    log: os.path.join(LOG_DIR, 'fastqc_{sample}_raw_R{read_num}.log')
+    params:
+        output_dir = os.path.join(FASTQC_DIR, '{sample}')
+    shell: "{FASTQC_EXEC} -o {params.output_dir} {input} >> {log} 2>&1"
+
+
+rule fastqc_trimmed:
+    input: os.path.join(TRIMMED_READS_DIR, "{sample}_trimmed_R{read_num}.fastq")
+    output: os.path.join(FASTQC_DIR, '{sample}', '{sample}_trimmed_R{read_num}_fastqc.zip')
+    log: os.path.join(LOG_DIR, 'fastqc_{sample}_trimmed_R{read_num}.log')
+    params:
+        output_dir = os.path.join(FASTQC_DIR, '{sample}')
+    shell: "{FASTQC_EXEC} -o {params.output_dir} {input} >> {log} 2>&1"
```

Comment:
This may or may not be a problem: does FastQC only produce the output zip?  Or are there more things it dumps in the output directory?  Snakemake will move on as soon as the expected output is generated, so if there's anything else in the output directory that we need we should declare it as an output.

Reply:
afaik when we tell it to output the zip, everything it will ever output is in that zip.

Reply:
It also outputs the html. I will make sure that both, html and zip is in the output section.

Reply:
:+1: